src/gdb/gdb.c

```diff
@@ -419,3 +446,41 @@ static void read_sr(int sr)
 	sregs[sr] = val;
 #endif
 }
+
+/* convert the hex array pointed to by buf into binary to be placed in mem
+ * return a pointer to the character after the last byte written
+ */
+static unsigned char *hex_to_mem(const unsigned char *buf, void *mem_,
+							int count)
+{
+	unsigned char *mem = mem_;
+	int i;
+	unsigned char ch;
+
+	if ((mem == NULL) || (buf == NULL))
+		return NULL;
+	for (i = 0; i < count; i++) {
+		ch = get_hex(*buf++) << 4;
+		ch |= get_hex(*buf++);
+#ifdef __XTENSA__
+	unsigned long tmp;
```

Comment:
This should call the xtensa specific function.

Reply:
There should be no xtensa specific code or macros in src/gdb/ whatsoever. There needs to be a arch/xtensa/gdb/core.c where arch_gdb_functionX() are defined and a arch/xtensa/include/arch/gdb.h where all macros are defined with inline assembly. e.g this function would call arch_gdb_hex_to_mem() macro defined in the arch gdb header.